components/modal/src/Modal.js

```diff
@@ -1,20 +1,33 @@
 /** @jsx jsx */
 
-import React, { Children, cloneElement, useState, useEffect } from 'react';
+import React, { createContext, useContext, useState, useEffect } from 'react';
 import { jsx } from '@westpac/core';
 import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import shortid from 'shortid';
 import FocusLock from 'react-focus-lock';
 import { CSSTransition } from 'react-transition-group';
-import { ModalBackdrop, StyledModal } from './styled';
-import { ModalHeader } from './ModalHeader';
-import { ModalBody } from './ModalBody';
+import { Backdrop, StyledModal } from './styled';
+
+// ==============================
+// Context and Consumer Hook
+// ==============================
+const ModalContext = createContext();
+
+export const useModalContext = () => {
+	const context = useContext(ModalContext);
+
+	if (!context) {
+		throw new Error('Modal sub-components should be wrapped in a <Modal>.');
+	}
+
+	return context;
+};
 
 // ==============================
 // Component
 // ==============================
-export const Modal = ({ isOpen, onClose, size, children, ...props }) => {
+export const Modal = ({ open: isOpen, onClose, size, children, ...props }) => {
 	const [open, setOpen] = useState(isOpen);
 
 	useEffect(() => {
```

Comment:
Hey shouldn't refer to the `open` state, rather than the `isOpen` prop?

```
useEffect(
		() => {
			setOpen(open);
		},
		[open]
	);
```

Reply:
No, since Im running the effect to update the internal state if the isOpen prop that is passed in changes. The modal should be controlled by the props being passed into it.